src/frontend/src/handler/create_table.rs

```diff
@@ -1182,20 +1181,25 @@ pub(super) async fn handle_create_table_plan(
                     let table: ExternalTableImpl = ExternalTableImpl::connect(config)
                         .await
                         .context("failed to auto derive table schema")?;
-                    let external_columns: Vec<_> = table
+                    let external_columns: HashMap<&str, ColumnCatalog> = table
                         .column_descs()
                         .iter()
-                        .cloned()
-                        .map(|column_desc| ColumnCatalog {
-                            column_desc,
-                            is_hidden: false,
+                        .map(|column_desc| {
+                            (
+                                column_desc.name(),
+                                ColumnCatalog {
+                                    column_desc: column_desc.clone(),
+                                    is_hidden: false,
+                                },
+                            )
                         })
                         .collect();
-                    for (col, external_col) in
-                        columns.iter_mut().zip_eq_fast(external_columns.into_iter())
-                    {
-                        col.column_desc.generated_or_default_column =
-                            external_col.column_desc.generated_or_default_column;
+
+                    for col in &mut columns {
+                        if let Some(external_col) = external_columns.get(col.name()) {
+                            col.column_desc.generated_or_default_column =
+                                external_col.column_desc.generated_or_default_column.clone();
+                        }
```

Comment:
Is it possible that we don't find a column in `external_columns`? Should we return an error (just like in #20203)?

Reply:
I cannot find a case for now. But a potential case is https://github.com/risingwavelabs/risingwave/pull/19112/#issuecomment-2613712659. 
An example haven't been supported but to be supported:
```
CREATE TABLE test (
    id int,
    name varchar,
    my_id int as id + 1, // generated column
    PRIMARY KEY (id)
) FROM pg_source TABLE 'public.test';
```
I can make it throw an error for now.

Reply:
Updated

Reply:
The same logic can be found [here](https://github.com/risingwavelabs/risingwave/blob/08bdf2743469d7aab35c98403f53a3f4738b93f5/java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java#L223), so it's OK not to throw an error, either.

Reply:
Sorry, I just recheck the implementation. If we don't throw an error here. Generated columns can be then supported in tables on cdc source. Previously, only tables with connectors can support generated columns. So I reverted the codes back and now it won't throw errors in this case.

Reply:
> So I reverted the codes back and now it won't throw errors in this case.

Shall we instead assert it's a generated column if it's not found?

Reply:
> Shall we instead assert it's a generated column if it's not found?

We can throw an error if a not-found column is not a generated column. 
It's also OK we don't do that, as it would also be checked in the [java code](https://github.com/risingwavelabs/risingwave/blob/08bdf2743469d7aab35c98403f53a3f4738b93f5/java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/common/PostgresValidator.java#L223).

Reply:
> If we don't throw an error here. Generated columns can be then supported in tables on cdc source. 

And I found the implementation here was buggy. The newly inserted rows are not received in this type of table. I'll try fixing it soon. If the fix is non-trivial, I throw an error here and leave the fix for future.

Reply:
> If the fix is non-trivial, I throw an error here and leave the fix for future.

I just reverted the codes. I'll open another PR to support generated columns in table on cdc source.